src/api.js

```diff
@@ -0,0 +1,82 @@
+const axios = require('axios');
+const rateLimit = require('axios-rate-limit');
+require('dotenv').config({path: '../.env'})
+const { ETHERSCAN_API_KEY, ETHERSCAN_API_URL, FAUCET_ADDRESS, GOERLI_API_URL } = process.env;
+
+const getBlockNumber = async function(time) {
+    const request = await rateLimit(axios.create(), {maxRequests: 5, perMillisecondss: 1000, maxRPS: 1})
+    const url = `${GOERLI_API_URL}?module=block&action=getblocknobytime&timestamp=${time}&closest=before&apikey=${ETHERSCAN_API_KEY}`
```

Comment:
Instead of building the same request url string on every call, in all functions mentioned in upper comment you can just use pre-built constants which can be defined on a file level scope or even somewhere in config file outside of this file.